superset/assets/src/explore/components/controls/MetricsControl.jsx

```diff
@@ -224,8 +224,13 @@ export default class MetricsControl extends React.PureComponent {
         (option.column_name.toLowerCase().indexOf(valueAfterAggregate.toLowerCase()) >= 0);
     }
     return option.optionName &&
-      (!option.metric_name || !this.isAutoGeneratedMetric(option)) &&
```

Comment:
@GabeLoins I'm trying to understand `!option.metric_name` why filtering this out?

Reply:
in the metrics control you can select saved_metrics, columns, or aggregates from the dropdown. That's why I use `optionName` for comparison rather than metric name. In that check I'm saying `if the option is not a saved metric,  or if it is a saved metric but not an autogenerated saved metric`